src/python/BuildingControlsSimulator/DataClients/DataClient.py

```diff
@@ -48,267 +51,137 @@ def make_data_directories(self):
-                                            identifier
-                                        ].spec.datetime_column
-                                    ]
-                                    >= p_start
-                                )
-                                & (
-                                    self.hvac[identifier].data[
-                                        self.hvac[
-                                            identifier
-                                        ].spec.datetime_column
-                                    ]
-                                    <= p_end
-                                )
-                            ]
-                        )
-
-                        sim_weather_data[identifier].append(
-                            self.weather[identifier].data[
-                                (
-                                    self.weather[identifier].data[
-                                        self.weather[
-                                            identifier
-                                        ].spec.datetime_column
-                                    ]
-                                    >= p_start
-                                )
-                                & (
-                                    self.weather[identifier].data[
-                                        self.weather[
-                                            identifier
-                                        ].spec.datetime_column
-                                    ]
-                                    <= p_end
-                                )
-                            ]
-                        )
-
-        return sim_hvac_data, sim_weather_data
+        # check for missing data sources
+        if all([len(d) != 0 for d in data_source_periods]):
+            # set period start of simulation time
+            p_start = self.sim_config["start_utc"]
+            p_end = self.sim_config["end_utc"]
+            # create list of data source idxs to keep track of place for each
+            ds_idx = [0 for d in data_source_periods]
+            data_periods = []
+            end_time = np.min([d[-1][1] for d in data_source_periods])
+            while p_start < end_time:
+                ds_p_start = []
+                ds_p_end = []
+
+                for i, d in enumerate(data_source_periods):
+                    if ds_idx[i] < len(d) - 1 and p_start >= d[ds_idx[i]][1]:
+                        # increment ds idx if period start is past end of ds period
+                        ds_idx[i] += 1
+
+                    ds_p_start.append(d[ds_idx[i]][0])
+                    ds_p_end.append(d[ds_idx[i]][1])
+
+                p_start = np.max(ds_p_start)
+                p_end = np.min(ds_p_end)
+                self.full_data_periods.append((p_start, p_end))
+                # advance time period
+                p_start = p_end
+
+            # select data only within periods
```

Comment:
Dead code?

Reply:
yes this is being refactored so data periods dont make copies of the input data along because the model initialization only happens once per simulation model (instead of once per simulation model - data period)